File: Javas/src/Problem3.java
import java.util.HashSet;

public class Problem3 implements Problem {


    private int[] numbers;
    private int exp;

    public Problem3(int[] numbers, int expected)
    {
        this.numbers = numbers;
        this.exp = expected;
    }



    @Override
    public Boolean Calculate() {
        int counter = 0;
        HashSet<Integer> set = new HashSet<Integer>();
        for (int num : numbers)
        {
            if (num == counter + 1)
            {
                counter++;
                while (set.contains(counter + 1))
                {
                    counter++;
                }
            }
            else
            {
                set.add(num);
            }
        }

        return counter+1 == this.exp;
    }
}

File: Scalas/src/main/scala/Problem3.scala
package problem

import scala.collection.mutable

class Problem3(list: List[Int], k: Int) extends Problem {
  override def Calculate(): Boolean = {

    var counter = 0
    var Set = new mutable.HashSet[Int]()

    for (elem <- list) {

      if ( elem == counter+1){
        counter += 1
        while (Set.contains(counter+1)){
          counter +=1
        }


      }
      else
        {
          Set.add(elem)
        }

    }
    return counter+1 == k
  }
}
